jbpm-event-emitters/jbpm-event-emitters-kafka/src/main/java/org/jbpm/event/emitters/kafka/KafkaEventEmitter.java

```diff
@@ -0,0 +1,161 @@
+            String type;
+            String topic;
+            if (view instanceof ProcessInstanceView) {
+                ProcessInstanceView processInstanceView = (ProcessInstanceView) view;
+                topic = "processes";
+                type = "process";
+                processInstanceId = processInstanceView.getId();
+                processId = processInstanceView.getProcessId();
+            } else if (view instanceof TaskInstanceView) {
+                TaskInstanceView taskInstanceView = (TaskInstanceView) view;
+                topic = "tasks";
+                type = "task";
+                processInstanceId = taskInstanceView.getProcessInstanceId();
+                processId = taskInstanceView.getProcessId();
+            } else if (view instanceof CaseInstanceView) {
+                CaseInstanceView caseInstanceView = (CaseInstanceView) view;
+                topic = "cases";
+                type = "case";
+                processInstanceId = caseInstanceView.getId();
+                processId = caseInstanceView.getCaseDefinitionId();
+            } else {
+                throw new UnsupportedOperationException("Unsupported view type " + view.getClass());
+            }
+            try {
+                producer.send(new ProducerRecord<>(getTopic(topic), mapper.writeValueAsBytes(
+                        new CloudEventSpec1(type, String.format(SOURCE_FORMATTER, processId, processInstanceId),
+                                view))));
+            } catch (IOException ex) {
+                throw new IllegalArgumentException("Error creating cloud event for view " + view, ex);
+            }
+        }
+    }
+
+    @SuppressWarnings({"squid:S2139", "squid:S3457"})
+    public void apply(Collection<InstanceView<?>> data) {
+        try {
+            localProducer.get().commitTransaction();
+        } catch (KafkaException ex) {
+            logger.error("Error publishing events " + data, ex);
+            localProducer.get().abortTransaction();
+            throw ex;
+        } finally {
+            localProducer.remove();
+        }
+    }
+
+    public void drop(Collection<InstanceView<?>> data) {
+        try {
+            localProducer.get().abortTransaction();
+        } finally {
+            localProducer.remove();
+        }
+    }
+
+    @Override
+    public void close() {
+        localProducer.remove();
+    }
+
+    @Override
+    public EventCollection newCollection() {
+        return new BaseEventCollection();
+    }
+
+    protected Producer<String, byte[]> getProducer() {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, System.getProperty(
+                "org.kie.jbpm.event.emitters.kafka.boopstrap.servers", "localhost:9092"));
+        configs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+        configs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, UUID.randomUUID().toString());
```

Comment:
Missing ProducerConfig.CLIENT_ID_CONFIG property. 
From Confluent documentation: _Although not required, you should always set a client.id since this allows you to easily correlate requests on the broker with the client instance which made it._